src/main/java/org/broadinstitute/hellbender/tools/spark/pipelines/metrics/CollectBaseDistributionByCycleSpark.java

```diff
@@ -0,0 +1,240 @@
+    public boolean pfReadsOnly = false;
+
+    private static final class HistogramGenerator implements Serializable{
+        private static final long serialVersionUID = 1L;
+
+        private static final int NBASES = 5; //ACGT + unknown
+        private int maxLengthSoFar = 0;
+        private final long[][] firstReadTotalsByCycle = new long[NBASES][maxLengthSoFar];
+        private long[] firstReadCountsByCycle = new long[maxLengthSoFar];
+        private final long[][] secondReadTotalsByCycle = new long[NBASES][maxLengthSoFar];
+        private long[] secondReadCountsByCycle = new long[maxLengthSoFar];
+        private boolean seenSecondEnd = false;
+
+        /**
+         * Merges the argument into this histogram generator. Returns the modified 'this' object.
+         * Note: you can only merge HistogramGenerator is they have the same 'useOriginalQualities' value.
+         */
+        public HistogramGenerator merge(final HistogramGenerator hg2) {
+            Utils.nonNull(hg2);
+            ensureArraysBigEnough(hg2.maxLengthSoFar);
+            for (int i = 0; i < NBASES; i++) {
+                for (int j = 0; j < hg2.firstReadTotalsByCycle[i].length; j++) {
+                    this.firstReadTotalsByCycle[i][j] += hg2.firstReadTotalsByCycle[i][j];
+                }
+                for (int j = 0; j < hg2.secondReadTotalsByCycle[i].length; j++) {
+                    this.secondReadTotalsByCycle[i][j] += hg2.secondReadTotalsByCycle[i][j];
+                }
+            }
+            for (int i = 0; i < hg2.firstReadCountsByCycle.length; i++) {
+                this.firstReadCountsByCycle[i] += hg2.firstReadCountsByCycle[i];
+            }
+            for (int i = 0; i < hg2.secondReadCountsByCycle.length; i++) {
+                this.secondReadCountsByCycle[i] += hg2.secondReadCountsByCycle[i];
+            }
+            this.seenSecondEnd = this.seenSecondEnd || hg2.seenSecondEnd;
+            return this;
+        }
+
+        private int baseToInt(final byte base) {
+            switch (base) {
+                case 'A':
+                case 'a':
+                    return 0;
+                case 'C':
+                case 'c':
+                    return 1;
+                case 'G':
+                case 'g':
+                    return 2;
+                case 'T':
+                case 't':
+                    return 3;
+            }
+            return 4;
+        }
+
+        HistogramGenerator addRead(final GATKRead rec) {
+            final byte[] bases = rec.getBases();
+            if (bases == null) {
+                return this;
+            }
+            final int length = bases.length;
+            final boolean rc = rec.isReverseStrand();
+            ensureArraysBigEnough(length + 1);
+            if (rec.isPaired() && rec.isSecondOfPair()) {
+                seenSecondEnd = true;
+                for (int i = 0; i < length; i++) {
+                    final int cycle = rc ? length - i : i + 1;
```

Comment:
It looks like the zero element is never populated, which is slightly weird. I guess it's so you don't have to map index 0 to cycle one?